maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/SurefireHelper.java

```diff
@@ -142,7 +142,10 @@ public static void reportExecution( SurefireReportParameters reportParameters, R
                                         PluginConsoleLogger log, Exception firstForkException )
         throws MojoFailureException, MojoExecutionException
     {
-        if ( firstForkException == null && !result.isTimeout() && result.isErrorFree() )
+        boolean isError = firstForkException != null || result.isTimeout() || !result.isErrorFree();
+        boolean isFlaky = reportParameters.getFailOnFlakeCount() > 0
+            && result.getFlakes() >= reportParameters.getFailOnFlakeCount();
```

Comment:
minor: could we have `getFailOnFlakeCount` on the same side on both cases to simplify comparison?

Reply:
@paplorinc 
What you mean?

Reply:
Just an observation that `X > 0 && X < Y` may be easier to read than `X > 0 && Y >= X`.
Don't have strong feelings about it, just an observation

Reply:
@paplorinc 
I think `result.getFlakes() >= reportParameters.getFailOnFlakeCount()` is okay because it is a condition saying `variable < constant`.